src/contradictory_claims/claims/extract_claims.py

```diff
@@ -0,0 +1,102 @@
+"""Function to extract Claims"""
+
+import pandas as pd
+import numpy as np
+import os
+
+from tqdm import tqdm
+import torch
+from torch.nn import ModuleList, Linear
+import torch.nn.functional as F
+from nltk import word_tokenize, sent_tokenize
+
+from allennlp.models.archival import load_archive
+from allennlp.predictors import Predictor
+from allennlp.common.file_utils import cached_path
+from allennlp.common.util import JsonDict
+from allennlp.data import Instance
+from allennlp.modules import Seq2VecEncoder, TimeDistributed, TextFieldEmbedder, ConditionalRandomField, FeedForward
+
+from .utils import read_json
+from .utils import ClaimAnnotationReaderJSON, CrfPubmedRCTReader
+from .utils import DiscourseClassifierPredictor
+from .utils import ClaimCrfPredictor
+
+
+
+def load_model(MODEL_PATH:str = "https://s3-us-west-2.amazonaws.com/pubmed-rct/model_crf.tar.gz", WEIGHT_PATH:str  = "https://s3-us-west-2.amazonaws.com/pubmed-rct/model_crf_tf.th"):
+    """This loads the weight from the path specified
+    
+    :param MODEL_PATH: location of model path, can be downloaded offline or link can be given
+    :param WEIGHT_PATH: location of model weight path, can be downloaded offline or link can be given, default link is specified
+    
+    :return: the model using the WEIGHT_PATH specified
+    """
+
+    archive = load_archive(MODEL_PATH) 
+    predictor = Predictor.from_archive(archive, 'discourse_crf_predictor')
+    # archive_ = load_archive("./model_crf.tar.gz")
+    # discourse_predictor = Predictor.from_archive(archive_, 'discourse_crf_predictor')
+    model = predictor._model
+    for param in list(model.parameters()):
+        param.requires_grad = False ## not train weights
+    EMBEDDING_DIM = 300
+    num_classes, constraints, include_start_end_transitions = 2, None, False
+    model.crf = ConditionalRandomField(num_classes, constraints, 
+                                    include_start_end_transitions=include_start_end_transitions)
+    model.label_projection_layer = TimeDistributed(Linear(2 * EMBEDDING_DIM, num_classes))
+    model.load_state_dict(torch.load(cached_path(WEIGHT_PATH), map_location='cpu'))
```

Comment:
what are these extra parameters/methods doing for the model? Some comments could help clarify